openedx_learning/apps/authoring/units/api.py

```diff
@@ -80,19 +79,17 @@ def create_unit_version(
     """
     with atomic():
         container_version = publishing_api.create_container_version(
-            unit.container.pk,
+            unit.pk,
             version_num,
             title=title,
             publishable_entities_pks=publishable_entities_pks,
             entity_version_pks=entity_version_pks,
             created=created,
             created_by=created_by,
         )
-        unit_version = UnitVersion.objects.create(
-            unit=unit,
-            container_version=container_version,
-            publishable_entity_version=container_version.publishable_entity_version,
-        )
+        # unit_version = UnitVersion(container_version=container_version)  # 🛑 This has a bug :/
+        unit_version = UnitVersion(container_version_id=container_version.pk)
+        unit_version.save_base(raw=True)  # 🛑 private method, but calling just .save() or .create() causes a bug :/
```

Comment:
I couldn't really figure out why the "normal" code here wasn't working, but I couldn't get `unit_version = UnitVersion(container_version=container_version)` or any simple variation of it to work, and had to resort to using the private `.save_base` method.

Reply:
Does Django require that we create the subclass directly, i.e. `UnitVersion(version_num, title, etc..)`

Reply:
How else would I create it? It's totally possible I'm missing something simple here.

Reply:
The first commented out line should actually say `unit_version = UnitVersion.objects.create(container_version=container_version)` because that's what I was trying without success.

Reply:
I mean, maybe you can't create the superclass first, and you have to just directly create only the subclass with all the fields that are in the superclass?

Reply:
Oh 🤦🏻. That does seem to work. I thought it was going to break our separation of concerns, but I realized it actually makes things even nicer, as long as I parametrize the publishing API methods like `create_container`.

Now it's mostly working and much cleaner. I'll keep cleaning this up and see where it goes, but I do like how it's looking.